### ISSUE TITLE
PDAL CLI - TileDB

### ISSUE THREAD

**Issue by normanb** [PDAL/PDAL#2768]
The PDAL CLI cannot parse a TileDB array as it doesn't have a file extension hence it is necessary to override the driver. This causes the following scenarios

```
pdal info --driver readers.tiledb sample_array 
PDAL: readers.tiledb: Missing value for positional argument 'array_name'

pdal info --driver readers.tiledb --readers.tiledb.array_name=sample_array
PDAL: No input file specified.

usage: pdal info [options] [input]
:
:
```

I would expect to be able to issue

```pdal info --driver readers.tiledb sample_array```

or

```pdal info --driver readers.tiledb --readers.tiledb.array_name=sample_array```

I can take a look at this in the near future. In the TileDB case (is this a more general problem with the CLI?) the `array_name` parameter is positional and shouldn't need to be named.

**Comment by abellgithub** [comment]
`pdal info` passes the positional argument on as `filename`.  The easiest fix is to add an alternate name for `array_name` as `filename`, which makes it consistent with other readers.  You can do this with the call on ProgramArgs called `addSynonym`.

**Comment by abellgithub** [comment]
I forgot that `filename` (unfortunately) is already defined for readers, so addSynonym will generate an error as it attempts to define it a second time.  I'm not sure of the best solution.  `array_name` could be added as a synonym for `filename`, but then you'd lose the nice description for `array_name` and the code would have to be changed to reference m_filename instead of m_arrayName.

The good solution IMO is removing the filename argument from the Reader and placing it in individual readers, but that touches lots of code.

Other ideas appreciated.

**Comment by abellgithub** [comment]
Fixed with 124061dde7fc8d80dae9a5d2fff94e80e138457c
